### PR TITLE
fix: Update rescoring filter if option was removed

### PR DESCRIPTION
**Pull request by zkdev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: src/components/dependencies/RescoringModal.js [hunk 1]
```diff
@@ -427,6 +427,15 @@ const RescoringFilterOption = ({
   const theme = useTheme()
   const [selected, setSelected] = React.useState([])
 
+  React.useEffect(() => {
+    // remote selected filter if option is not available anymore (e.g. last rescoring was deleted)
```
Comment (8R0WNI3) [inline src/components/dependencies/RescoringModal.js:431]: nit: `remote` -> `remove`

**Review by 8R0WNI3 — DISMISSED** [review]
/lgtm

File: src/components/dependencies/RescoringModal.js [hunk 2]
```diff
@@ -427,6 +427,15 @@ const RescoringFilterOption = ({
...
+    if (selected.every((s) => options.some((o) => o.name === s))) return
+    // ensure to only update state if required
```
Comment (ccwienk) [inline src/components/dependencies/RescoringModal.js:433]: I think the placement of this comment is somewhat misleading, as it refers to the line above, does it not?

File: src/components/dependencies/RescoringModal.js [hunk 3]
```diff
@@ -427,6 +427,15 @@ const RescoringFilterOption = ({
   const theme = useTheme()
   const [selected, setSelected] = React.useState([])
 
+  React.useEffect(() => {
+    // remote selected filter if option is not available anymore (e.g. last rescoring was deleted)
+    if (selected.every((s) => options.some((o) => o.name === s))) return
```
Comment (ccwienk) [inline src/components/dependencies/RescoringModal.js:432]: I think this is quite hard to read. How about constructing a set of selected names, and check whether the set of available option-names is a subset instead?

Comment (zkdev) [reply src/components/dependencies/RescoringModal.js:432]: oh wow, there is [Set.isSubsetOf()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/isSubsetOf) 🤯 

> Baseline 2024 Newly available

It appears this feature was released quite recently.

Will update, but for the record:
We will rather have to check whether selected is subset of options, not the other way around.

**Pull request merged**